src/main/java/com/shima/patientchart/mapper/PatientChartMapper.java

```diff
@@ -1,14 +1,44 @@
 package com.shima.patientchart.mapper;
 
 import com.shima.patientchart.entity.PatientChart;
-import org.apache.ibatis.annotations.Mapper;
-import org.apache.ibatis.annotations.Select;
+import org.apache.ibatis.annotations.*;
 
 import java.util.List;
+import java.util.Optional;
 
 @Mapper
 public interface PatientChartMapper {
 
+    //全DBの情報を取得
     @Select("SELECT * FROM patientcharts")
     List<PatientChart> findAll();
+
+    //GET　指定したid
+    @Select("SELECT * FROM patientcharts WHERE id = #{id}")
+    //患者カルテ選択からどこでID
+    Optional<PatientChart> findById(int id);
+
+
+    //POST処理 (新規追加登録処理)
+    @Insert("INSERT INTO patientcharts (name, gender, address, insurancecard, medicalhistory) VALUES (#{name},#{gender},#{address}, #{insurancecard}, #{medicalhistory})")
+    @Options(useGeneratedKeys = true, keyProperty = "id")
+    void insert(PatientChart patientChart);
+
+    //POST バリデーション1　Address
+    @Select("SELECT * FROM patientcharts WHERE address = #{address}")
+    Optional<PatientChart> findByAddress(String address);
+
+    //POST バリデーション2 Name
+    @Select("SELECT * FROM patientcharts WHERE name = #{name}")
+    Optional<PatientChart> findByName(String name);
+
+    @Select("SELECT * FROM patientcharts WHERE address = #{address} AND id != #{id}")
+    Optional<PatientChart> findByAddressExcept(String address, int id);
```

Comment:
`findByAddressExcept`だと何を除外しているのかわからないので、idが除外されることがわかるように、`findByAddressExceptId`にしてあげましょう。
```suggestion
    Optional<PatientChart> findByAddressExceptId(String address, int id);
```

Reply:
修正を行いました。

### 動作確認
既存DBにない id をしてPATCH処理を行った際


curl --location --request PATCH 'http://localhost:8080/patient-charts/8' \
--header 'Content-Type: application/json' \
--data '{
    "address": "佐賀",
    "insurancecard": "国保",
    "medicalhistory": "胃潰瘍"
}'



![8D7FBBC3-2AF3-4B10-97CD-2E831DDAC829](https://github.com/simamumu89/Kadai10/assets/142807674/516d3c3b-64d7-402b-820b-979bb7ec4961)


Addressの重複ロジック


curl --location --request PATCH 'http://localhost:8080/patient-charts/6' \
--header 'Content-Type: application/json' \
--data '{
    "address": "鹿児島",
    "insurancecard": "国保",
    "medicalhistory": "十二指腸潰瘍"
}'



![4B45384D-5312-4CBE-86AF-5F9DEE3552F7](https://github.com/simamumu89/Kadai10/assets/142807674/f75920cc-06b0-43c1-a185-8ba1f7a8a205)